### PR TITLE
First draft of iterative ai design principles

### PR DESCRIPTION
**Pull request by testobsessed**
Not sure if we want to merge this as is or iterate on it on a branch for a while. I'm fine either way.

### CODE REVIEW

File: iterative-ai-principles.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+## Naming
+We need a name for this document that in some way telegraphs the benefit of the approach or the intention behind it.
+
+My original name was "modern AI" but modern is the wrong word for something that is so nascent. 
+
+Roman and Tanya suggested "composable," and that's a better placeholder name.
+
+In this draft I'm using "iterative" but I'm not sure that's right either.
+
+I suspect we need a different word to describe this. I just don't know what it is yet. Suggestions & edits welcome.
```
Comment (dadarstan) [inline iterative-ai-principles.md:19]: I think both "iterative" and "composable" deliver the massage of flexibility, however the former is flexibility in time and the latter in structure (initially it referred to infrastructure, not process, but could refer to a process, too). Both make sense in my mind. I was also thinking about the word "emerging" but that might be too VC of me :)

Comment (Sadosh42) [reply iterative-ai-principles.md:19]: "Emerging" has different than "composable" and "iterative" meaning for me - actually two last words also have quite different meaning for me. I keep thinking, what do AI stack architects want? What can we give them, from what they want? They probably want the infrastructure to enable outcomes of better responses, faster and more efficient processing, perhaps simply unlocking new capabilities. If so, the word "composable" seems a better fit to me, as composability allows both different iterations and emerging of new building blocks (new stuff can emerge out of different compositions)

File: iterative-ai-principles.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+## Principles
+The intent behind these principles is to create AI systems that can be iteratively developed and engender trust. To that end:
```
Comment (dadarstan) [inline iterative-ai-principles.md:22]: Shall we start with some very basic principles of openness and control? I'm not asking for full definition of open source AI, but we can state requirements for 
- ability to control the choice of the model and level of its observability (e.g. is training data available)
- private data staying local (local inference) 
- ability to change model (i.e. open weights)

Comment (rvs) [reply iterative-ai-principles.md:22]: Purely in the interest of brainstorming (@testobsessed please pick the bits that you like -- the following is just a suggestion) -- we may want to refer to the "essential freedoms" in this document as well (perhaps as just a link?) https://www.gnu.org/philosophy/free-sw.html.en#clarifying

**Comment by rvs** [comment]
Just to unblock this -- I suggest we merge the original proposal by @testobsessed as-is and then have separate pull requests addressing all the comments raised.

**Pull request merged**